content/lua/concepts/strings/terms/find/find.md

```diff
@@ -0,0 +1,176 @@
+
+## Example 1
+
+Use `.find()` without multiple assignments:
+
+```lua
+sentence = 'hello world'
+search1 = string.find(sentence, "hello")
+search2 = string.find(sentence, "world")
+search3 = string.find(sentence, "universe")
+
+print(search1)
+print(search2)
+print(search3)
+```
+
+The above example assigns the function to a unique variable, in order for the results to be returned/printed to the console.
+
+The results of the above functions are:
+
+```shell
+1
+7
+nil
+```
+
+- `hello` is found at starting index 1 in the `sentence` string.
+- `world` is found at starting index 7 in the `sentence` string.
+- `universe` is not found in `sentence` string, therefore the output is nil.
+
+## Example 2
+
+Use `.find()` with multiple assignments:
+
+```lua
+sentence = "hello world"
+a, b = string.find(sentence, "hello")
+print(a)
+print(b)
+
+print(a, b)
+
+y, z = string.find(sentence, "world")
+print(y)
+print(z)
+
+print(y, z)
+```
+
+The above example uses multiple assignment, to save the results to individual variables, in order for the results to be returned/printed to the console.
+
+The results of the above functions are:
+
+```shell
+1
+5
+1  5
+7
+11
+7  11
+```
+
+- `hello` starts at index 1, and ends at index 5 in the `sentence` string. Due to multiple assignments, `a` saves the value `1`, and `b` saves the second value `5`.
+- `world` starts at index 7, and ends at index 11 in the `sentence` string. Due to multiple assignments, `y` saves the first value `7`, and `z` saves the second value `11`.
+
+## Example 3
+
+Use `.find()` with `init` (Initial position) parameter:
+
+```lua
+sentence1 = 'hello world'
+search = string.find(sentence1, "l", 5)
+print(search)  -- Output: 10
+
+sentence2 = 'world world'
+a, b = string.find(sentence2, "orl", 5)
+print(a)  -- Output: 8
+print(b)  -- Output: 10
+
+print(a, b)  -- Output: 8  10
+```
+
+The above example uses the initial position parameter to indicate where the search shout start.
+
+The results of the above functions are:
+
+```shell
+10
+8
+10
+8  10
+```
+
+- `l` has multiple instances within the `sentence` string. Due to the init parameter, the search starts at index 5, therefore the first instance of `l` from index 5 will be at starting index `10`.
+- `orl` has multiple instances within the `sentence` string. Due to the init parameter, the search starts at index 5, therefore the first instance of `orl` from index 5 will be at starting index `8`, and ends at index `10`.
+
+## Example 4
+
+Use `.find()` with `pattern` (pattern matching) parameter:
+
+```lua
+-- With pattern matching
+sentence = "hello world"
+a, b = string.find(sentence, "l+", 1, false)
+print(a)
+print(b)
+
+-- Without pattern matching
+y, x = string.find("hello world", "l+", 1, true)
+print(y)
+print(z)
```

Comment:
```suggestion
sentence = "hello world"
a, b = string.find(sentence, "l+", 1, false)
print(a)
print(b)

y, x = string.find("hello world", "l+", 1, true)
print(y)
print(z)
```

Reply:
Won't having the comments distinguish between the code with pattern matching and the code without pattern matching make quickly telling the difference between the two easier? without it, you would either have to explicitly indicate the difference in the explanation on lines 173 and 174, or hope the user remembers the difference, or require them to scroll up.